frappe/core/doctype/sms_settings/sms_settings.py

```diff
@@ -66,16 +66,18 @@ def send_sms(receiver_list, msg, sender_name = '', success_msg = True):
 def send_via_gateway(arg):
 	ss = frappe.get_doc('SMS Settings', 'SMS Settings')
 	headers = get_headers(ss)
+	use_json = "Content-Type" in headers and headers["Content-Type"] == "application/json"
 
-	args = {ss.message_parameter: arg.get('message')}
+	message = arg.get('message').decode('utf-8') if use_json else arg.get('message')
```

Comment:
let's use safe_decode, then you don't need to check for `"Content-Type"` in headers either
```suggestion
	message = frappe.safe_decode(arg.get('message'))
```